cs/src/core/Device/LocalStorageDevice.cs

```diff
@@ -53,14 +57,19 @@ private void RecoverFiles()
 
             string bareName = fi.Name;
 
-            int prevSegmentId = -1;
+            List<int> segids = new List<int>();
             foreach (FileInfo item in di.GetFiles(bareName + "*"))
             {
-                int segmentId = Int32.Parse(item.Name.Replace(bareName, "").Replace(".", ""));
+                segids.Add(Int32.Parse(item.Name.Replace(bareName, "").Replace(".", "")));
+            }
+            segids.Sort();
+
+            int prevSegmentId = -1;
+            foreach (int segmentId in segids)
+            {
                 if (segmentId != prevSegmentId + 1)
                 {
                     startSegment = segmentId;
-
                 }
                 else
                 {
```

Comment:
Do we need to truncate the segments that are not tracked by this device?

Reply:
You mean delete existing files when the device is created? That would depend upon whether the device is being used as new or for recovery.